diff -uprN configglue.orig/pyschema/options.py configglue/pyschema/options.py
--- configglue.orig/pyschema/options.py	2010-12-16 16:57:47.000000000 -0300
+++ configglue/pyschema/options.py	2010-12-15 16:44:00.000000000 -0300
@@ -74,7 +74,10 @@ class LinesConfigOption(ConfigOption):
             else:
                 value = self.item.parse(value, raw=raw)
             return value
-        items = [_parse_item(x) for x in value.split('\n') if len(x)]
+        if type(value) != list:
+	    items = [_parse_item(x) for x in value.split('\n') if len(x)]
+	else:
+	    items = value
         if self.remove_duplicates:
             filtered_items = []
             for item in items:
