diff -upr django.orig/utils/translation/trans_real.py django/utils/translation/trans_real.py
--- django.orig/utils/translation/trans_real.py	2009-03-18 13:55:59.000000000 -0300
+++ django/utils/translation/trans_real.py	2010-08-09 15:03:24.000000000 -0300
@@ -344,11 +344,14 @@ def get_language_from_request(request):
     from django.conf import settings
     globalpath = os.path.join(os.path.dirname(sys.modules[settings.__module__].__file__), 'locale')
     supported = dict(settings.LANGUAGES)
-
-    if hasattr(request, 'session'):
-        lang_code = request.session.get('django_language', None)
-        if lang_code in supported and lang_code is not None and check_for_language(lang_code):
-            return lang_code
+    
+    try:
+	if hasattr(request, 'session'):
+    	    lang_code = request.session.get('django_language', None)
+    	    if lang_code in supported and lang_code is not None and check_for_language(lang_code):
+        	return lang_code
+    except:
+	pass
 
     lang_code = request.COOKIES.get(settings.LANGUAGE_COOKIE_NAME)
     if lang_code and lang_code in supported and check_for_language(lang_code):
