diff -uprN django_cpserver.orig/management/commands/runcpserver.py django_cpserver/management/commands/runcpserver.py
--- django_cpserver.orig/management/commands/runcpserver.py	2010-11-01 18:40:53.000000000 -0300
+++ django_cpserver/management/commands/runcpserver.py	2010-11-01 17:52:31.000000000 -0300
@@ -3,57 +3,27 @@
 import logging, sys, os, signal, time, errno
 from socket import gethostname
 from django.core.management.base import BaseCommand
-
+from django.conf import settings
 
 CPSERVER_HELP = r"""
   Run this project in a CherryPy webserver. To do this, CherryPy from
   http://www.cherrypy.org/ is required.
 
-   runcpserver [options] [cpserver settings] [stop]
-
-Optional CherryPy server settings: (setting=value)
-  host=HOSTNAME         hostname to listen on
-                        Defaults to localhost
-  port=PORTNUM          port to listen on
-                        Defaults to 8088
-  server_name=STRING    CherryPy's SERVER_NAME environ entry
-                        Defaults to localhost
-  daemonize=BOOL        whether to detach from terminal
-                        Defaults to False
-  pidfile=FILE          write the spawned process-id to this file
-  workdir=DIRECTORY     change to this directory when daemonizing
-  threads=NUMBER        Number of threads for server to use
-  ssl_certificate=FILE  SSL certificate file
-  ssl_private_key=FILE  SSL private key file
-  server_user=STRING    user to run daemonized process
-                        Defaults to www-data
-  server_group=STRING   group to daemonized process
-                        Defaults to www-data
-
-Examples:
-  Run a "standard" CherryPy server server
-    $ manage.py runcpserver
-
-  Run a CherryPy server on port 80
-    $ manage.py runcpserver port=80
-
-  Run a CherryPy server as a daemon and write the spawned PID in a file
-    $ manage.py runcpserver daemonize=true pidfile=/var/run/django-cpserver.pid
-
+   runcpserver [options] [stop]
 """
 
 CPSERVER_OPTIONS = {
-'host': 'localhost',
-'port': 8088,
-'server_name': 'localhost',
-'threads': 10, 
-'daemonize': False,
-'workdir': None,
-'pidfile': None,
-'server_user': 'www-data',
-'server_group': 'www-data',
-'ssl_certificate': None,
-'ssl_private_key': None,
+'host': settings.CP_HOST,
+'port': settings.CP_PORT,
+'server_name': settings.CP_SERVER_NAME,
+'threads': settings.CP_THREADS,
+'daemonize': settings.CP_DAEMONIZE,
+'workdir': settings.CP_WORKDIR,
+'pidfile': settings.CP_PIDFILE,
+'server_user': settings.CP_SERVER_USER,
+'server_group': settings.CP_SERVER_GROUP,
+'ssl_certificate': settings.CP_SSL_CERTIFICATE,
+'ssl_private_key': settings.CP_SSL_PRIVATE_KEY,
 }
 
 
