diff -uprN django.orig/db/models/sql/query.py django/db/models/sql/query.py
--- django.orig/db/models/sql/query.py	2009-11-26 10:18:29.000000000 -0300
+++ django/db/models/sql/query.py	2009-11-26 10:33:51.000000000 -0300
@@ -2177,7 +2177,10 @@ class BaseQuery(object):
                 entry_params = []
                 pos = entry.find("%s")
                 while pos != -1:
-                    entry_params.append(param_iter.next())
+            	    # allow %s to be sent to the database, example calling strftime
+            	    # from sqlite
+            	    if entry.find("%%s", pos-1) != pos-1:
+                	entry_params.append(param_iter.next())
                     pos = entry.find("%s", pos + 2)
                 select_pairs[name] = (entry, entry_params)
             # This is order preserving, since self.extra_select is a SortedDict.
