diff -uprN PyOBEX//common.py PyOBEX.mine//common.py
--- PyOBEX//common.py		2010-06-12 16:47:24.000000000 -0300
+++ PyOBEX.mine//common.py	2010-05-31 16:32:48.000000000 -0300
@@ -141,34 +113,7 @@ class Message:
 
 
 class MessageHandler:
-
-    format = ">BH"
-    
-    if sys.platform == "win32":
-    
-        def _read_packet(self, socket_):
-        
-            data = ""
-            while len(data) < 3:
-                data += socket_.recv(3 - len(data))
-            type, length = struct.unpack(self.format, data)
-            while len(data) < length:
-                data += socket_.recv(length - len(data))
-            return type, length, data
-    else:
-    
-        def _read_packet(self, socket_):
-        
-            data = socket_.recv(3, socket.MSG_WAITALL)
-            type, length = struct.unpack(self.format, data)
-            if length > 3:
-                data += socket_.recv(length - 3, socket.MSG_WAITALL)
-            return type, length, data
-    
-    def decode(self, socket):
-    
-        code, length, data = self._read_packet(socket)
-        
+    def decode(self, code, length, data):
         if self.message_dict.has_key(code):
             message = self.message_dict[code]()
             message.read_data(data)
diff -uprN PyOBEX//responses.py PyOBEX.mine//responses.py
--- PyOBEX//responses.py	2010-06-12 16:47:24.000000000 -0300
+++ PyOBEX.mine//responses.py	2010-08-18 11:18:13.000000000 -0300
@@ -82,10 +82,7 @@ class ResponseHandler(MessageHandler):
     
     UnknownMessageClass = UnknownResponse
     
-    def decode_connection(self, socket):
-    
-        code, length, data = self._read_packet(socket)
-        
+    def decode_connection(self, code, length, data):
         if code == ConnectSuccess.code:
             message = ConnectSuccess()
         elif self.message_dict.has_key(code):
