diff -uprN timezones/fields.py timezones/fields.py
--- timezones/fields.py	2010-07-28 20:41:12.000000000 -0300
+++ timezones/fields.py	2010-08-09 11:13:13.000000000 -0300
@@ -155,3 +155,8 @@ def prep_localized_datetime(sender, **kw
 ## RED_FLAG: need to add a check at manage.py validation time that
 ##           time_zone value is a valid query keyword (if it is one)
 signals.class_prepared.connect(prep_localized_datetime)
+
+from south.modelsinspector import add_introspection_rules
+add_introspection_rules([], [
+    "^timezones\.fields\.TimeZoneField",
+    "^timezones\.fields\.LocalizedDateTimeField",])
